frontend/apps/filemanager/filemanager.lua

```diff
@@ -274,14 +274,26 @@ function FileManager:setupLayout()
                     text = _("Open with…"),
                     callback = function()
                         UIManager:close(self.file_dialog)
-                        local one_time_providers = {
-                            {
-                                provider_name = _("Text viewer"),
+                        local one_time_providers = {}
+                        if DocSettings.image_ext[util.getFileNameSuffix(file):lower()] then
```

Comment:
(In case you didn't, please check this doesn't crash when a filename has no suffix.)

Reply:
Also a bit odd that the reference of supported images is now `DocSettings.image_ext` (DocSettings, in its name and features, does not related to the support of images).
But ok if you can't find a better place for that list.

Reply:
> please check this doesn't crash when a filename has no suffix

It's okay
https://github.com/koreader/koreader/blob/f7d633cc3d0efd273a875132ac2de4fb4754dd02/frontend/util.lua#L975

Reply:
> Also a bit odd that the reference of supported images is now `DocSettings.image_ext`

Maybe make it `util.image_ext`?
It is used by DocSettings, Filemanager, FilemanagerBookInfo, and can be added to Screensaver too.

Reply:
And in ImageWidget itself
https://github.com/koreader/koreader/blob/f7d633cc3d0efd273a875132ac2de4fb4754dd02/frontend/ui/widget/imagewidget.lua#L131-L134

Reply:
Well, "util" also feels odd for a supported image extensions list.
Looking at what is in frontend/ and frontend/ui/, there's no obvious module candidate to receive such definition.
Maybe frontend/ui/renderimage.lua ? Haven't looked if this module would somehow be used in the end by all the locations that need that list of supported images, and if it is sane to have it loaded just to get the list (looks like it is loaded by ImageWidget, so quite early, so it may be fine?)

Reply:
DocumentRegistry?